apps/src/StudioApp.js

```diff
@@ -733,61 +728,30 @@ StudioApp.prototype.loadBlocks = function(blocksXml) {
 *    block position.
 */
 StudioApp.prototype.arrangeBlockPosition = function(startBlocks, arrangement) {
-  var type, arrangeX, arrangeY;
+
+  var type, x, xmlChild;
+
   var xml = parseXmlElement(startBlocks);
-  var xmlChildNodes = this.sortBlocksByVisibility(xml.childNodes);
-  var numberOfPlacedBlocks = 0;
-  for (var x = 0, xmlChild; xmlChildNodes && x < xmlChildNodes.length; x++) {
+
+  var xmlChildNodes = xml.childNodes || [];
+  arrangement = arrangement || {};
+
+  for (x = 0; x < xmlChildNodes.length; x++) {
```

Comment:
oh, mostly that I originally had two for loops in here and didn't put it back in when I took the second one out. I'll go ahead and turn it into a `for (var i = )`